src/inference/tests/functional/ov_core_threading.cpp

```diff
@@ -169,17 +169,60 @@ TEST_F(CoreThreadingTests, GetAvailableDevices) {
 }
 
 #if defined(ENABLE_OV_IR_FRONTEND)
-// tested function: read_model and add_legacy_extension
+
+namespace ov {
+namespace test {
+namespace util {
+class Barrier {
+private:
+    std::mutex m_mutex;
+    std::condition_variable m_cv;
+    size_t m_count;
+    size_t m_expected;
+    size_t m_gen;
+
+public:
+    explicit Barrier(std::size_t count) : m_count{count}, m_expected{count}, m_gen{} {}
+
+    void arrive_and_wait() {
+        std::unique_lock<std::mutex> lock(m_mutex);
+        auto gen = m_gen;
+
+        if (!--m_count) {
```

Comment:
More readable?
```suggestion
        if (--m_count == 0) {
```

Reply:
If there will be any update will change if not clean it later.